packages/flutter/lib/src/services/keyboard.dart

```diff
@@ -75,6 +101,19 @@ class KeyboardHandle {
     assert(_keyboard._currentHandle != this);
   }
 
+  /// Changes the keyboard's state.
+  ///
+  /// The given `state` is uploaded to the keyboard, overwriting whatever state
+  /// the keyboard had previously.
+  ///
+  /// Interacting with the keyboard is inherently racy because there are two
+  /// writers: code that wishes to change keyboard state spontaneously (e.g.,
+  /// because the user pasted some text) and the user themselves. For this
```

Comment:
That doesn't make it racy. The reason it's racy is that there's two writers_and the whole system as asynchronous_.